### PR TITLE
Alpha masking module

### PR DESCRIPTION
**Pull request by blurry-x-face**
Concerns #165  

This PR adds Alpha Masking module!

Screenshot from 2020-01-20 19-43-22

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/is-reviewers` for help, in a comment below
* [x] Insert-step functionality is working correct as expected.
> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software
Please make sure to get at least two reviews before asking for merging the PR as that would make the PR more reliable on our part
Thanks!

### CODE REVIEW

**Comment by blurry-x-face** [comment]
@publiclab/is-reviewers please review this!

**Comment by blurry-x-face** [comment]
@jywarren tests added :smile:

File: src/modules/Mask/Module.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      UI.notify('Offset Unavailable', 'offset-notification');
+      callback();
+    }
+
+    const alpha_masking = function(c, a) {
+      return (a * c + (255 - a) * c) / 255;
+    };
+    const internalSequencer = ImageSequencer({ inBrowser: false, ui: false });
+    internalSequencer.loadImage(priorStep.output.src, function() {
+      internalSequencer.importJSON([{ 'name': 'resize', 'options': { resize: options.resize } }]);
```
Comment (HarshKhandeparkar) [inline src/modules/Mask/Module.js]: Can this be done without using an internal sequencer, please?  Internal sequencer has a ton of overhead. Try using the edge-detect approach (check to code to see how it uses Blur Module).

Comment (blurry-x-face) [reply src/modules/Mask/Module.js]: I tried that but wasn't able to do that.

Comment (HarshKhandeparkar) [reply src/modules/Mask/Module.js]: I don't think that can be done as of now. Refer: https://github.com/publiclab/image-sequencer/issues/1560

**P.S.**: If you use the current approach, it won't work with GIFs. Please add that to the end of the README.

Comment (blurry-x-face) [reply src/modules/Mask/Module.js]: Ok, So I have opened a pull request to refactor resize #1562

**Comment by jywarren** [comment]
This looks awesome. Catching up on the PRs now due to GitPod speeding up review! Thank you!!!

File: src/modules/Mask/info.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (jywarren) [inline src/modules/Mask/info.json:1]: Really nicely done. Well formatted, readable, excellent work!

**Comment by jywarren** [comment]
Just confirming this works great in GitPod!

image

**Comment by jywarren** [comment]
Noting #flat

> Hi, i'm so sorry, there is just one thing. Is it possible to use a flat file for the test? The inline data-url is just too gigantic. I think other tests show how to do this pretty well. Alternatively if you get really stuck, you could `require()` in the image from a separate file so the tests themselves are readable. Thank you!

Noting we can address this in https://github.com/publiclab/image-sequencer/issues/1502

This should be good!

**Review by jywarren — APPROVED** [review]
Nice! Getting ready to merge this.

**Review by HarshKhandeparkar — APPROVED** [review]
Great! Used the [Github CLI](https://cli.github.com). It is awesome!﻿

**Comment by jywarren** [comment]
Hooray!!!!!

**Pull request merged**